SOURCES/etc/xapi.d/plugins/smartctl.py

```diff
@@ -14,32 +14,47 @@ def _list_disks():
     disks = []
     result = run_command(['smartctl', '--scan'])
     for line in result['stdout'].splitlines():
-        if line.startswith('/dev/') and not line.startswith('/dev/bus/'):
-            disks.append(line.split()[0])
+        disks.append(line.split()[0])
```

Comment:
I think that you can get the type of the disk here. No need to an extra function _list_raids.
Maybe return a tuple here with line.split()[0] and line.split()[2]

Reply:
I'v tryed te solve that.

Reply:
I think the most readable way to do that is to use a dict. For example you can do:
`devices = {'name': line.split()[0], 'type': line.split()[2]}`
By doing this way we know what is expecting and when calling the `smartctl` later in the code you won't need the index you will be able to use the explicit field.

Reply:
Note that I used `devices` here because it looks like it is closer from the man page of the smartctl.

Reply:
And also renamed `_list_disks` into `_list_devices` if you decide to rename it.